modules/stochastic_tools/src/samplers/LatinHypercubeSampler.C

```diff
@@ -0,0 +1,109 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "LatinHypercubeSampler.h"
+#include "Distribution.h"
+
+registerMooseObjectAliased("StochasticToolsApp", LatinHypercubeSampler, "LatinHypercube");
+
+defineLegacyParams(LatinHypercubeSampler);
+
+InputParameters
+LatinHypercubeSampler::validParams()
+{
+  InputParameters params = Sampler::validParams();
+  params.addClassDescription("Latin Hypercube Sampler.");
+  params.addRequiredParam<dof_id_type>("num_rows",
+                                       "The number of rows per matrix to generate; a single value "
+                                       "or one value per distribution may be supplied.");
+  params.addRequiredParam<std::vector<DistributionName>>(
+      "distributions",
+      "The distribution names to be sampled, the number of distributions provided defines the "
+      "number of columns per matrix.");
+  params.addRequiredParam<std::vector<unsigned int>>(
+      "num_bins", "The number of intervals to consider within the sampling.");
+  params.addRequiredParam<std::vector<double>>(
+      "upper_limits", "The lower limit of probability for each of the associated distributions.");
+  params.addRequiredParam<std::vector<double>>(
+      "lower_limits", "The upper limit of probability for each of the associated distributions.");
+  return params;
+}
+
+LatinHypercubeSampler::LatinHypercubeSampler(const InputParameters & parameters)
+  : Sampler(parameters),
+    _distribution_names(getParam<std::vector<DistributionName>>("distributions")),
+    _num_bins_input(getParam<std::vector<unsigned int>>("num_bins")),
+    _upper_limits(getParam<std::vector<double>>("upper_limits")),
+    _lower_limits(getParam<std::vector<double>>("lower_limits"))
+{
+  for (const DistributionName & name : _distribution_names)
+    _distributions.push_back(&getDistributionByName(name));
+
+  if (_upper_limits.size() != _distributions.size())
+    paramError("upper_limits",
+               "The length of 'upper_limits' must match the length of 'distributions'");
+
+  if (_lower_limits.size() != _distributions.size())
+    paramError("lower_limits",
+               "The length of 'lower_limits' must match the length of 'distributions'");
+
+  if (_num_bins_input.size() != 1 && _num_bins_input.size() != _distributions.size())
+    paramError("num_bins",
+               "The length of 'num_bins' must be one or match the length of 'distributions'");
+
+  for (const auto & lim : _upper_limits)
```

Comment:
I'm 90% sure that range checking works for vectors. You might try it!

Reply:
I will try.

Reply:
It does support it, but the error messages for my version are a bit better. If I didn't already have them programmed and tested I would use it. Something for next time.